File: src/main/java/ru/sejapoe/digitalhotelserver/user/session/Session.java
package ru.sejapoe.digitalhotelserver.user.session;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import ru.sejapoe.digitalhotelserver.core.db.BaseDocument;
import ru.sejapoe.digitalhotelserver.core.security.BitArray256;
import ru.sejapoe.digitalhotelserver.user.User;

import java.time.LocalDateTime;

@Document
public class Session extends BaseDocument {
    @DBRef
    private User user;
    private BitArray256 sessionKey;
    @CreatedDate
    private LocalDateTime createdAt;

    public Session(User user, BitArray256 sessionKey) {
        this.user = user;
        this.sessionKey = sessionKey;
        this.createdAt = LocalDateTime.now();
    }

    public User getUser() {
        return user;
    }

    public BitArray256 getSessionKey() {
        return sessionKey;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
}

File: src/jvmMain/kotlin/ru/sejapoe/application/user/Session.kt
package ru.sejapoe.application.user

import org.jetbrains.exposed.dao.IntEntity
import org.jetbrains.exposed.dao.IntEntityClass
import org.jetbrains.exposed.dao.id.EntityID
import org.jetbrains.exposed.dao.id.IntIdTable
import org.jetbrains.exposed.sql.Table.Dual.default
import ru.sejapoe.application.utils.BitArray256
import java.time.LocalDateTime
import java.time.ZoneOffset

class Session(id: EntityID<Int>) : IntEntity(id) {
    companion object : IntEntityClass<Session>(Sessions)

    var user by User referencedOn Sessions.user
    var sessionKey by Sessions.sessionKey.transform({ it.bytes }, { BitArray256(it) })
    var createdAt: LocalDateTime by Sessions.createdAt.default(LocalDateTime.now().toEpochSecond(ZoneOffset.UTC))
        .transform({ it.toEpochSecond(ZoneOffset.UTC) }, { LocalDateTime.ofEpochSecond(it, 0, ZoneOffset.UTC) })
}

object Sessions : IntIdTable() {
    val user = reference("user", Users)
    val sessionKey = binary("session_key", 32)
    val createdAt = long("created_at")
}
